src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -56,6 +57,7 @@ public Command parseCommand(String userInput) throws ParseException {
 
         return switch (commandWord) {
         case AddCommand.COMMAND_WORD -> new AddCommandParser().parse(arguments);
+        case AddPublicAddressCommand.COMMAND_WORD -> new AddPublicAddressCommandParser().parse(arguments);
```

Comment:
Same thing here, need to add a testcase in addressbookparsertest for the case of addpublicaddress command word